### ISSUE TITLE
Initial checkout on system running vscode client?

### ISSUE THREAD

**Issue by chrmarti** [microsoft/vscode-remote-release#5667]
[...]

Edit: While testing podman (remote) and setting up a new vm for docker locally, I fell across an issue regarding VPN connections. To me it looks like you're doing an initial checkout looking for a devcontainer definition on the system running the vscode client. In both of my scenarios that machine has no access to my repository server. Is there an existing workaround for this? Is this intended? Do you want/need a new issue for this?

_Originally posted by @max06 in https://github.com/microsoft/vscode-remote-release/issues/5609#issuecomment-933318298_

**Comment by chrmarti** [comment]
@max06 With checkout you mean `git clone`? We clone the repository in a temporary container on the target system where you want your dev container to run.

**Comment by max06** [comment]
Mhm. When I use "Clone Repository in container Volume", the first call to git happens locally/in wsl.

```
[46427 ms] Start: Run: wsl -d Ubuntu -e /bin/sh -c cd '.' && git 'ls-remote' 'git@git.some-protected-server.internal:group/repo.git'
```

**Comment by chrmarti** [comment]
We locally check if the repository exists.

**Comment by max06** [comment]
That makes sense...

I know, I'm talking about rare edge cases here - the current implementation requires me to have two active vpn connections, the exact opposite of why I've built that ubuntu vm. This may work in some cases, but there are vpn servers out there only allowing one active connection per user.

While being stuck at this point I've also tried to make use of my socks5-proxy (ssh dynamic forward) which usually works perfectly fine in my chromium based browser. Without success, I wasn't able to redirect the git call at all.
